### PR TITLE
Changing permissions of UCP - removing unused permissions

### LINKED ISSUE #8082: Update and remove unnecessary permissions of UCP component

### PR DESCRIPTION
**Pull request by ytimocin**
# Description
Changing permissions of UCP - removing unused permissions.

## Type of change
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).
Fixes: #8082 

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- [ ] An overview of proposed schema changes is included in a linked GitHub issue.
- [ ] A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
- [ ] If applicable, design document has been reviewed and approved by Radius maintainers/approvers.
- [ ] A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
- [ ] A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
- [ ] A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.

### CODE REVIEW

File: deploy/Chart/templates/ucp/rbac.yaml [hunk 1]
```diff
@@ -7,51 +7,32 @@ metadata:
...
-- apiGroups:
-  - api.ucp.dev
-  resources:
-  - '*'
-  verbs:
-  - '*'
+    - create
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
```
Comment (ytimocin) [inline deploy/Chart/templates/ucp/rbac.yaml:38]: Removing this causes this error (an error that we have all seen before but didn't know why):

```
Radius release not found. Proceeding with installation.
*** Installing Radius to Kubernetes ***
Installing Radius version 8d1442d-dirty to namespace: radius-system...
*** Create workspace, group and environment for test ***
Creating workspace...
Set "kind-kind" as current workspace
Error: An unknown error was returned while testing Radius API status:
Status Code: 503
Response Headers:
  Audit-Id: 863b6ad6-02a3-4ba6-9c51-61f03dc66cf2
  Cache-Control: no-cache, private
  Content-Length: 20
  Content-Type: text/plain; charset=utf-8
  Date: Sun, 24 Nov 2024 16:36:20 GMT
  X-Content-Type-Options: nosniff
  X-Kubernetes-Pf-Flowschema-Uid: db65b004-bc64-4949-a3b2-2a0[80](https://github.com/radius-project/radius/actions/runs/11997783030/job/33443799631?pr=8080#step:18:81)1c4da00
  X-Kubernetes-Pf-Prioritylevel-Uid: fe22ed3a-e9f4-4f37-9901-d2347e099c15
Response Body:
service unavailable
```

File: deploy/Chart/templates/ucp/rbac.yaml [hunk 2]
```diff
@@ -6,52 +6,45 @@ metadata:
...
+      - delete
+      - get
+      - list
+      - update
+
+  - apiGroups:
+      - ucp.dev
+    resources:
+      - queuemessages
+      - resources
```
Comment (rynowak) [inline deploy/Chart/templates/ucp/rbac.yaml:34]: Resources is duplicated here. 

- https://github.com/radius-project/radius/pull/8080/files#diff-24ef6f7a71ac496e97b570c9d42f82358cbd496e660eb058079543cc05ff1805R34
- https://github.com/radius-project/radius/pull/8080/files#diff-24ef6f7a71ac496e97b570c9d42f82358cbd496e660eb058079543cc05ff1805R22

**Review by rynowak — APPROVED** [review]

File: deploy/Chart/templates/ucp/rbac.yaml (hunk 1 above)
Comment (kachawla) [reply deploy/Chart/templates/ucp/rbac.yaml:43]: @ytimocin did you get a chance to look into the logs for error?

Comment (ytimocin) [reply deploy/Chart/templates/ucp/rbac.yaml:43]: Unfortunately, I missed that window. It is a resource permission issue but not exactly sure which one. I will keep an eye out for this error.

**Pull request merged**